src/main/java/com/hazelcast/kubernetes/KubernetesClient.java

```diff
@@ -156,6 +155,18 @@ String zone(String podName) {
         return extractZone(callGet(nodeUrlString));
     }
 
+    /**
+     * Retrieves node name for the specified {@code namespace} and the given {@code podName}.
+     *
+     * @param podName POD name
+     * @return Node name
+     * @see <a href="https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11">Kubernetes Endpoint API</a>
+     */
+    String nodeName(String podName) {
```

Comment:
Could you use this method in `zone(String podName)`. Like the `zone` method should now look like this:
```
        String nodeUrlString = String.format("%s/api/v1/nodes/%s", kubernetesMaster, nodeName(podName));
        return extractZone(callGet(nodeUrlString));
```